app/views/shared/menu/_large_menu.html.haml

```diff
@@ -2,7 +2,11 @@
   %section.top-bar-section
     %ul.nav-logo
       %li.ofn-logo
-        %a{href: main_app.root_path}
+        - if @white_label_distributor&.preferred_white_label_logo_link.present?
+          - link = @white_label_distributor.preferred_white_label_logo_link 
+        - else
+          - link = main_app.root_path
```

Comment:
This kind of logic is better in a helper.

Reply:
I've changed that: 837e062ad3